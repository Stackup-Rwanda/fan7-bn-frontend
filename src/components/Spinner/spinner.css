.loading {
  width: 100px;
  height: 100px;
  margin: 22% auto;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  border-radius: 50%;
  border-top: 5px solid #A2AAB1;
  position: relative;
  vertical-align: middle;
  -webkit-animation: a1 2s linear infinite;
          animation: a1 2s linear infinite;
}

.loading::before, .loading::after {
  content: '';
  width: 100px;
  height: 100px;
  position: absolute;
  left: 0;
  top: -5px;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  border-top: 5px solid #A2AAB1;
  border-radius: 50%;
}

.loading::before {
  -webkit-transform: rotate(120deg);
          transform: rotate(120deg);
}

.loading::after {
  -webkit-transform: rotate(240deg);
          transform: rotate(240deg);
}

.loading span {
  position: absolute;
  width: 100px;
  height: 100px;
  color: #000000;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  text-align: center;
  line-height: 100px;
  -webkit-animation: a2 2s linear infinite;
          animation: a2 2s linear infinite;
}

@-webkit-keyframes a1 {
  to {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}

@keyframes a1 {
  to {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}

@-webkit-keyframes a2 {
  to {
    -webkit-transform: rotate(-360deg);
            transform: rotate(-360deg);
  }
}

@keyframes a2 {
  to {
    -webkit-transform: rotate(-360deg);
            transform: rotate(-360deg);
  }
}
/*# sourceMappingURL=spinner.css.map */