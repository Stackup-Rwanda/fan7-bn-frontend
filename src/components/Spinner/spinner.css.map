{
    "version": 3,
    "mappings": "AAAA,AAAA,QAAQ,CAAC;EACL,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,MAAM,EAAE,QAAQ;EAChB,UAAU,EAAE,UAAU;EACtB,aAAa,EAAE,GAAG;EAClB,UAAU,EAAE,iBAAiB;EAC7B,QAAQ,EAAE,QAAQ;EAClB,cAAc,EAAE,MAAM;EACtB,SAAS,EAAE,qBAAqB;CAiCnC;;AA1CD,AAWI,QAXI,AAWH,QAAQ,EAXb,QAAQ,AAYH,OAAO,CAAC;EACL,OAAO,EAAE,EAAE;EACX,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,QAAQ,EAAE,QAAQ;EAClB,IAAI,EAAE,CAAC;EACP,GAAG,EAAE,IAAI;EACT,UAAU,EAAE,UAAU;EACtB,UAAU,EAAE,iBAAiB;EAC7B,aAAa,EAAE,GAAG;CACrB;;AAtBL,AAwBI,QAxBI,AAwBH,QAAQ,CAAC;EACN,SAAS,EAAE,cAAc;CAC5B;;AA1BL,AA4BI,QA5BI,AA4BH,OAAO,CAAC;EACL,SAAS,EAAE,cAAc;CAC5B;;AA9BL,AAgCI,QAhCI,CAgCJ,IAAI,CAAC;EACD,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,OAAO;EACd,UAAU,EAAE,UAAU;EACtB,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,KAAK;EAClB,SAAS,EAAE,qBAAqB;CACnC;;AAGL,UAAU,CAAV,EAAU;EACN,EAAE;IACE,SAAS,EAAE,cAAc;;;;AAIjC,UAAU,CAAV,EAAU;EACN,EAAE;IACE,SAAS,EAAE,eAAe",
    "sources": [
        "spinner.scss"
    ],
    "names": [],
    "file": "spinner.css"
}