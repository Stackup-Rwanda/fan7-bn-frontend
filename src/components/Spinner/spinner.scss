.loading {
    width: 100px;
    height: 100px;
    margin: 22% auto;
    box-sizing: border-box;
    border-radius: 50%;
    border-top: 5px solid #A2AAB1;
    position: relative;
    vertical-align: middle;
    animation: a1 2s linear infinite;

    &::before, 
    &::after {
        content: '';
        width: 100px;
        height: 100px;
        position: absolute;
        left: 0;
        top: -5px;
        box-sizing: border-box;
        border-top: 5px solid #A2AAB1;
        border-radius: 50%;
    }

    &::before {
        transform: rotate(120deg);
    }

    &::after {
        transform: rotate(240deg);
    }

    span {
        position: absolute;
        width: 100px;
        height: 100px;
        color: #000000;
        box-sizing: border-box;
        text-align: center;
        line-height: 100px;
        animation: a2 2s linear infinite;
    }
}

@keyframes a1 {
    to {
        transform: rotate(360deg);
    }
}

@keyframes a2 {
    to {
        transform: rotate(-360deg);
    }
}