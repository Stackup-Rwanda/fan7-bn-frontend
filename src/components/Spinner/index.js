import React from 'react';
import './spinner.scss';

const Loading = () => (
  <div className="loading">
    <span>Loading...</span>
  </div>
);

export default Loading;
