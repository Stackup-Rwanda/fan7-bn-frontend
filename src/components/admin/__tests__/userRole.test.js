import React from 'react';
import { mount, shallow } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import UserRole from '../userRole';
import Spinner from '../../Spinner/index';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const history = {
  push: jest.fn(),
  replace: jest.fn(),
  createHref: jest.fn(),
};

describe('UserRole Settings Component Test', () => {
  beforeEach(() => {
    localStorage.setItem('barefoot_nomad_token', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTcsInNvY2lhbF9pZCI6IjExMzQzOTIwNjYyNDI1NTM0NzA2NiIsInVzZXJuYW1lIjoiTWlzdGljbyBDbGVtZW50IiwiaW1hZ2UiOiJodHRwczovL2xoMy5nb29nbGV1c2VyY29udGVudC5jb20vYS0vQU9oMTRHZ01DX0lUc096b251WFhEVDU4a0J3aWpoaHlIeS1sa0wyTUpTR2hzZyIsInByb3ZpZGVyIjoiZ29vZ2xlIiwiaWF0IjoxNTg2Mjc1NTQzfQ.EqpaqDDAUdPXryiczJPsRxylfSVQts4t9Nhm5AUZPhM');
  });
    it('should return true if component exists', () => {
        const initialState = {
          chat: {
            messages: {
              senderId: 1,
              sender: 'The Great',
              message: 'Test react',
              createdAt: '01-03-2019'
            }
          },
            role: { loading: false,
                users: [{
                    email: "e.kskemc@gmail.com",
                    role: "requester"
                }],
            },

        };
        const store = mockStore(initialState);
        const wrapper = mount(
          <Provider store={store}>
            <Router history={history}>
              <UserRole />
            </Router>
          </Provider>,
        );
    
        expect(wrapper.find('.body_display').exists()).toBe(true);
      });

    it('should display spinner if it\'s loading', () => {
        const initialState = {
          chat: {
            messages: {
              senderId: 1,
              sender: 'The Great',
              message: 'Test react',
              createdAt: '01-03-2019'
            }
          },
            role: { loading: true,
                users: [{
                    email: "e.kskemc@gmail.com",
                    role: "requester"
                }],
            },
        };
        const store = mockStore(initialState);
        const wrapper = mount(
          <Provider store={store}>
            <Router history={history}>
              <UserRole />
            </Router>
          </Provider>,
        );
    
        expect(wrapper.find(Spinner).exists()).toBe(true);
      });

      it('should display content if it is not loading', () => {
        const initialState = {
          chat: {
            messages: {
              senderId: 1,
              sender: 'The Great',
              message: 'Test react',
              createdAt: '01-03-2019'
            }
          },
          role: { loading: false,
            users: [{
                email: "e.kskemc@gmail.com",
                role: "requester"
            }],
         },
        };
        const store = mockStore(initialState);
        const wrapper = mount(
          <Provider store={store}>
            <Router history={history}>
              <UserRole />
            </Router>
          </Provider>,
        );
    
        expect(wrapper.find('.body_display_table').exists()).toBe(true);
        expect(wrapper.find('.body_display_pagination').exists()).toBe(true);
    });
});

describe('<UserRole /> shallow rendering tests', () => {
  beforeEach(() => {
    localStorage.setItem('barefoot_nomad_token', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTcsInNvY2lhbF9pZCI6IjExMzQzOTIwNjYyNDI1NTM0NzA2NiIsInVzZXJuYW1lIjoiTWlzdGljbyBDbGVtZW50IiwiaW1hZ2UiOiJodHRwczovL2xoMy5nb29nbGV1c2VyY29udGVudC5jb20vYS0vQU9oMTRHZ01DX0lUc096b251WFhEVDU4a0J3aWpoaHlIeS1sa0wyTUpTR2hzZyIsInByb3ZpZGVyIjoiZ29vZ2xlIiwiaWF0IjoxNTg2Mjc1NTQzfQ.EqpaqDDAUdPXryiczJPsRxylfSVQts4t9Nhm5AUZPhM');
  });
    const initialState = {
        role: { loading: false,
            users: [{
                email: "e.kskemc@gmail.com",
                role: "requester"
            }],
        },

    };
    const store = mockStore(initialState);
  
    it('matches the snapshot', () => {
      const tree = shallow(<UserRole store={store} />);
      expect(tree).toMatchSnapshot();
    });
});

describe('Should handle Dropdown',() => {
  beforeEach(() => {
    localStorage.setItem('barefoot_nomad_token', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTcsInNvY2lhbF9pZCI6IjExMzQzOTIwNjYyNDI1NTM0NzA2NiIsInVzZXJuYW1lIjoiTWlzdGljbyBDbGVtZW50IiwiaW1hZ2UiOiJodHRwczovL2xoMy5nb29nbGV1c2VyY29udGVudC5jb20vYS0vQU9oMTRHZ01DX0lUc096b251WFhEVDU4a0J3aWpoaHlIeS1sa0wyTUpTR2hzZyIsInByb3ZpZGVyIjoiZ29vZ2xlIiwiaWF0IjoxNTg2Mjc1NTQzfQ.EqpaqDDAUdPXryiczJPsRxylfSVQts4t9Nhm5AUZPhM');
  });
    
    it('Should display Dropdown  ', () => {
        const initialState = {
          chat: {
            messages: {
              senderId: 1,
              sender: 'The Great',
              message: 'Test react',
              createdAt: '01-03-2019'
            }
          },
            role: { loading: false,
                users: [{
                    email: "e.kskemc@gmail.com",
                    role: "requester"
                }],
            },
        };
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
              <Router history={history}>
                <UserRole />
              </Router>
            </Provider>,
          );
  
    });
});