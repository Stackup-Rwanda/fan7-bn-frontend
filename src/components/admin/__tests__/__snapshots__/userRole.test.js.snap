// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRole /> shallow rendering tests matches the snapshot 1`] = `
<ContextProvider
  value={null}
>
  <_class
    dispatch={[Function]}
    loading={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    users={
      Array [
        Object {
          "email": "e.kskemc@gmail.com",
          "role": "requester",
        },
      ]
    }
  />
</ContextProvider>
`;
